lib/src/common/repos_service.dart

```diff
@@ -968,10 +968,31 @@ class RepositoriesService extends Service {
 
   /// Fetches a single release by the release tag name.
   ///
+  /// Returns null if the release doesn't exist.
+  ///
   /// API docs: https://developer.github.com/v3/repos/releases/#get-a-release-by-tag-name
-  Future<Release> getReleaseByTagName(RepositorySlug slug, String tagName) =>
-      github.getJSON('/repos/${slug.fullName}/releases/tags/$tagName',
-          convert: (i) => Release.fromJson(i));
+  Future<Release> getReleaseByTagName(
+      RepositorySlug slug, String tagName) async {
+    // github.getJSON('/repos/${slug.fullName}/releases/tags/$tagName', convert: (i) => Release.fromJson(i));
+
+    Release release;
+    try {
+      release = await github.getJSON(
+        '/repos/${slug.fullName}/releases/tags/$tagName',
+        convert: (i) => Release.fromJson(i),
+        statusCode: StatusCodes.OK,
+        fail: (http.Response response) {
+          if (response.statusCode == 404) {
+            // we just return null if the tag can't be found.
+            throw ReleaseNotFound.fromTagName(github, tagName);
+          }
+        },
+      );
+    } on ReleaseNotFound catch (e) {
+      release = null;
```

Comment:
If I'm reading this correctly, this would mean that if a release isn't found callers might just get a null and not really know why. The exception is handled internally and silenced.
If that's correct it would mean it behaves inconsistently with other methods. For example the getReleaseById method would throw an exception if the release doesn't exist whereas this method returns null.

Would you be open to simply allowing the ReleaseNotFound exception to come out of this function and handling the exception at the caller level in your consuming code?

Reply:
Consistency is probably the right answer.

Personally I don't think an exception should be thrown unless something 'exceptional' happens.

I would consider not finding a release a part of normal processing rather than an exceptional set of circumstances.

The resulting code is also easier to work with

if ((release = getReleaseByTagname(xxx)) != null)
{ 
   release.delete();
}

as apposed to
try
{
   var release = getReleaseByTagName(xxxx);
   release.delete();
}
on ReleaseNotFound catch(e)
{
/// no op
}

Reply:
I think I'd prefer the consistency with other methods and not return null. If someone did want to handle the case when a release is not found .. they wouldn't be able to. Here's what I was thinking it should look like:
```dart
  Future<Release> getReleaseByTagName(
      RepositorySlug slug, String tagName) async {
    return github.getJSON(
      '/repos/${slug.fullName}/releases/tags/$tagName',
      convert: (i) => Release.fromJson(i),
      statusCode: StatusCodes.OK,
      fail: (http.Response response) {
        if (response.statusCode == 404) {
          // we just return null if the tag can't be found.
          throw ReleaseNotFound.fromTagName(github, tagName);
        }
      },
    );
  }
```

Reply:
OK, I have modified the code as requested. I will run a couple of tests and push the code.